### PR TITLE
Add installation instructions and citation to README

### PR DESCRIPTION
**Pull request by mschmidt87**
Add installation instructions to README. 
Note: I noticed that installing extra requirements is tedious at the moment. See #144 
Add citation and DOI badge to README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,6 +31,26 @@ While the `ConstantFloat` node and derived nodes have a fixed output value, the
 The library provides a local search function that performs stochastic gradient descent on the values of `Parameter` nodes (cf. Topchy & Punch, 2001; Izzo et al., 2017).
 See `examples/example_differential_evo_regression.py` for an example evolving an expression containing mathematical constants.
 
+Installation
+============
+
+To install the release version, simply execute
```
Comment (jakobj) [inline README.md:37]:
```suggestion
You can install the latest relase via pip:
```

File: README.md [hunk 2]
```diff
@@ -28,6 +31,26 @@ While the `ConstantFloat` node and derived nodes have a fixed output value, the
...
+```shell script
+pip install hal-cgp
+```
+
+This library comes with some extra requirements, defined in `extra_requirements.txt`. For instance, if you want to parse an individual to a `sympy` expression or a `torch` object, you'll need to install the extra requirements by running
```
Comment (jakobj) [inline README.md:42]:
```suggestion
This library depends on some optional packages defined in `extra_requirements.txt`. These are necessary, for example, to compile an individual to a SymPy expression or a PyTorch class. You can install all extra requirements via:
```

File: README.md [hunk 3]
```diff
@@ -28,6 +31,26 @@ While the `ConstantFloat` node and derived nodes have a fixed output value, the
...
+```shell script
+pip install hal-cgp[all]
+```
+
+To install the current master, run
```
Comment (jakobj) [inline README.md:47]:
```suggestion
The adventurous can install the most recent development version directly from our master branch (don't use this in production unless there are good reasons!):
```

**Comment by mschmidt87** [comment]
Thanks, I implemented your suggestions and squashed the commits.

**Pull request merged**